### PR TITLE
add pulled bwa for alignment and add concat task

### PR DESCRIPTION
**Pull request by m28feng**

### CODE REVIEW

**Comment by m28feng** [comment]
retest this please

**Comment by m28feng** [comment]
retest this please

**Comment by m28feng** [comment]
retest this please

**Comment by m28feng** [comment]
retest this please

**Comment by m28feng** [comment]
retest this please

**Comment by m28feng** [comment]
retest this please

**Review by lheisler — APPROVED** [review]
this looks good.
my understanding that will be provided two arrays of fastq files, for R1 and R2.  It will concatenate each array to a single fastq, saving with a prefix that is provided.

concatenated fastq are aligned through the bwa mem workflow

the rest is the same, providing the alignment to ichorcna instead of using an alignment provided by the olive

**Comment by m28feng** [comment]
retest this please

**Review by lheisler — APPROVED** [review]

File: commands.txt [hunk 1]
```diff
@@ -0,0 +1,74 @@
+## Commands
+This section lists command(s) run by WORKFLOW workflow
```
Comment (pruzanov) [inline commands.txt:2]: As I mentioned elsewhere, this is a placeholder (WORKFLOW needs to be changed to ichorCNA)

File: README.md [hunk 2]
```diff
@@ -95,31 +131,81 @@ Output | Type | Description
...
-
-## Support
+## Commands
+ This section lists command(s) run by WORKFLOW workflow
+ 
+ * Running WORKFLOW
+ 
+ IchorCNA allows for quantification of tumor content in cfDNA. The input for this workflow is an array of fastq pairs with their read group information. This ichorCNA workflow first calls bwaMem for an alignment to the specified reference genome; then if multiple fastq pairs are specified the bam files are merged using samtools. The next step prepares the data for ichorCNA which is the final step in the workflow.
+ 
+ MERGE BAMS
```
Comment (pruzanov) [inline README.md:141]: Also, all this changes should be listed in CHANGELOG.md file

**Review by pruzanov — CHANGES_REQUESTED** [review]
I see that only minor changes are needed, the code looks good to me, not sure if you want to change calculate.sh script (it uses line count, not md5sum) under the current PR

**Comment by bibilujan** [comment]
@pruzanov the changes to the calculate.sh will be applied in the near future, but will move forward with this development for now as suggested by @lheisler .

**Pull request merged**